staging/src/k8s.io/cloud-provider/credentialconfig/provider.go

```diff
@@ -14,10 +14,9 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package credentialprovider
```

Comment:
Personally in favor of keeping this file in `pkg/credentialprovider` and keeping this PR scoped to only moving the GCP plugin to staging. The credential provider logic is very kubelet specific and probably doesn't belong in the `k8s.io/cloud-provider` module.

Reply:
Aside from that LGTM after commits are squashed :)

Reply:
I agree that the DockerConfigProvider interface does not really belong in k8s.io/cloud-provider ... long-term, isn't the intent to use https://github.com/kubernetes/enhancements/issues/2133 for GCP to provide credentials? would that mean moving the credential provider interface/types into staging under k8s.io/kubelet instead?

Reply:
> would that mean moving the credential provider interface/types into staging under k8s.io/kubelet instead?

Yeah which was why we had moved the necessary types/APIs to staging in the initial implementation (https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/kubelet/pkg/apis/credentialprovider). IIRC you should be able to build a credential provider with only those types and without depending on anything in `k8s.io/kubernetes/pkg/credentialprovider`.

Reply:
right. that makes more sense to me than moving the credential provider interface and types to k8s.io/cloud-provider

Reply:
I recall @cheftako mentioning a good reason for doing this though, maybe I'm missing context